test/integration/testutil/kubectl.go

```diff
@@ -22,15 +24,22 @@ type Kubectl struct {
 
 // Command is a general func to run kubectl commands.
 func (k *Kubectl) Command(cmdOptions ...string) ([]byte, error) {
-	return k.executeKubectl(nil, cmdOptions...)
+	return k.executeKubectl(context.Background(), nil, cmdOptions...)
 }
 
 // CommandWithInput is a general func to run kubectl commands with stdin input.
 func (k *Kubectl) CommandWithInput(stdinInput string, cmdOptions ...string) ([]byte, error) {
-	return k.executeKubectl(strings.NewReader(stdinInput), cmdOptions...)
+	return k.executeKubectl(context.Background(), strings.NewReader(stdinInput), cmdOptions...)
 }
 
-func (k *Kubectl) executeKubectl(stdIn io.Reader, cmdOptions ...string) ([]byte, error) {
+func (k *Kubectl) CommandWithTimeout(timeout time.Duration, cmdOptions ...string) ([]byte, error) {
```

Comment:
While this might be good enough for now, I would expose a method with a context e.g.
```go
func (k *Kubectl) CommandContext(ctx context.Context, cmdOptions ...string) ([]byte, error) { }
```
so it's more reusable (like fixing the [`contextcheck` lint issue](https://github.com/Twingate/kubernetes-access-gateway/pull/96/files#r2255661722)). I think it's common to have method with context as its first argument in Go.